tests/cli.py

```diff
@@ -785,5 +785,27 @@ def test_ip_leases_no_nm_lease(self):
         self.assertIn(b'No lease found', err)
         self.assertNotEqual(p.returncode, 0)
 
+    def test_ip_leases_generate_mapping_is_empty(self):
+        mock_generate = MockCmd('generate')
+        # When the interface in not defined in any netplan yaml file
+        # the command 'generate --mapping <interface>' will return nothing
+        mock_generate.set_output('')
+        os.environ.setdefault('NETPLAN_GENERATE_PATH', mock_generate.path)
+
+        c = os.path.join(self.workdir.name, 'etc', 'netplan')
+        os.makedirs(c)
```

Comment:
I guess we could/should move this up into the `setUp` method of `class TestIp`, as that might be generically useful to more tests (in the future).

Reply:
Agree. I will do that in another PR not related to this bug fix.